gateway/src/main/java/io/zeebe/gateway/impl/broker/cluster/BrokerClusterStateImpl.java

```diff
@@ -46,18 +48,22 @@ public BrokerClusterStateImpl(BrokerClusterStateImpl topology) {
 
   public BrokerClusterStateImpl() {
     partitionLeaders = new Int2IntHashMap(NODE_ID_NULL);
+    partitionLeaderTerms = new Int2ObjectHashMap<>();
     partitionFollowers = new Int2ObjectHashMap<>();
     brokerAddresses = new Int2ObjectHashMap<>();
     brokers = new IntArrayList(5, NODE_ID_NULL);
     partitions = new IntArrayList(32, PARTITION_ID_NULL);
     randomBroker = new Random();
   }
 
-  public void setPartitionLeader(int partitionId, int leaderId) {
-    partitionLeaders.put(partitionId, leaderId);
-    final List<Integer> followers = partitionFollowers.get(partitionId);
-    if (followers != null) {
-      followers.removeIf(follower -> follower == leaderId);
+  public void setPartitionLeader(int partitionId, int leaderId, long term) {
+    if (partitionLeaderTerms.getOrDefault(partitionId, -1L) < term) {
+      partitionLeaders.put(partitionId, leaderId);
+      partitionLeaderTerms.put(partitionId, Long.valueOf(term));
```

Comment:
This is not necessary

Reply:
We have to keep track of the terms, no? Otherwise, we wouldn't be able to order the updates.

Reply:
Sorry I mean the Long.valueOf(term) which is a long :sweat_smile:

Reply:
ah.. IDE is complaining without it.

Reply:
:laughing: